File: proto-java/src/main/java/com/spaceape/protobuf/reflect/FloatFieldDescriptor.java
package com.spaceape.protobuf.reflect;

import com.google.protobuf.CodedInputStream;
import com.google.protobuf.CodedOutputStream;
import com.spaceape.protobuf.CodeGenContext;

import java.io.IOException;

public class FloatFieldDescriptor extends FieldDescriptor {
    public FloatFieldDescriptor(int fieldNumber, String fieldName, Class clazz){
        super(fieldNumber, fieldName, clazz, float.class);
    }
    public Float readField(CodedInputStream input, CodeGenContext context) throws IOException {
        return input.readFloat();
    }
    public void writeField(CodedOutputStream output, CodeGenContext context, Object object) throws IOException{
        output.writeFloat(fieldNumber, (float)object);
    }
}

File: proto-csharp/ProtoMessage/ProtoMessage/ApeMessage/Reflect/FloatFieldDescriptor.cs
﻿using System;
using System.Reflection;
using System.Collections.Generic;
using Google.ProtocolBuffers;
using com.spaceape.protobuf;

namespace com.spaceape.protobuf.reflect
{
  public class FloatFieldDescriptor: FieldDescriptor
  {
    public FloatFieldDescriptor(int fieldNumber, String fieldName, Type clazz) : base(fieldNumber, fieldName, clazz)
    {
    }

    override public Object readField(CodedInputStream input, CodeGenContext context)
    {
      return input.ReadFloat();
    }

    override public void writeField(CodedOutputStream output, CodeGenContext context, Object obj)
    {
      output.WriteFloat(fieldNumber, (float)obj);
    }
  }
}

